### PR TITLE
Compare View Revamp: TR-55 Runoff Charts and Tables

### PR DESCRIPTION
**Pull request by rajadain**
## Overview

Updated version of #2123. For more granular albeit noisy commits, see here: https://github.com/WikiWatershed/model-my-watershed/compare/develop...tt+ki/compare-view-tr55-runoff

Connects #2072 

## Testing Instructions

 * Check out this branch
 * Open a saved project, or create a new one, using TR-55 as the model and at least 2 scenarios
 * Open the Compare View
 * Ensure the values are correct in both the chart and the table
 * Move the precipitation slider, ensure the values update correctly
 * Ensure that the values of the original scenarios do not change after closing the Compare View

### CODE REVIEW

File: src/mmw/js/src/core/chart.js [hunk 1]
```diff
@@ -314,8 +314,86 @@ function renderLineChart(chartEl, data, options) {
...
+             .reduceXTicks(false)
+             .staggerLabels($svg.width() < widthCutoff)
+             .duration(0)
+             .margin(options.margin)
+             .color(colors)
+             .showXAxis(false)
+             .id(name)
+             .yDomain([0, yDomainMax]);
+
+
```
Comment (kellyi) [inline src/mmw/js/src/core/chart.js]: I accidentally added 2 extra whitespace lines here; probably worth dropping them.

File: src/mmw/js/src/compare/models.js [hunk 2]
```diff
@@ -1,11 +1,80 @@
...
+        this.scenarios = options.scenarios;
+
+        this.scenarios.forEach(function(scenario) {
+            scenario.get('results').on('change', update);
+        });
+    }
+});
+
+var Tr55RunoffCharts = ChartRowsCollection.extend({
+    update: function() {
```
Comment (arottersman) [inline src/mmw/js/src/compare/models.js]: Noice.

**Comment by kellyi** [comment]
Taking a look.

**Comment by kellyi** [comment]
This is looking good! We can continue to think about this but I kind of like the idea of dropping the top yDomain value, since it's sometimes very close in spatial terms to the existing tick:

screen shot 2017-08-15 at 10 22 06 am

Also I think we may wanna adjust the `tickCount` setting here -- https://github.com/WikiWatershed/model-my-watershed/pull/2148/files#diff-3e33aa79d87dd300905a2a0b6c714e56R351 -- or nearby, as it looks like it's not catching every case in which the values are small:

screen shot 2017-08-15 at 10 24 49 am

I think `if (maxValue <= 0.2)` *should* work there.

**Comment by kellyi** [comment]
Checked this out in Chrome, FF, Safari, and IE11. Works well in Chrome & FF; but the charts & table don't display in either IE11 or Safari. Looks like the DOM elements are there, but they're not rendering visibly:

screen shot 2017-08-15 at 10 39 33 am

I imagine this has something to do with the CSS rules we're applying to the row elements?

**Comment by rajadain** [comment]
What is the desired behavior for the ticks? I'm not sure what changing the value to `0.2` wants to achieve.

**Comment by kellyi** [comment]
> I'm not sure what changing the value to 0.2 wants to achieve.

Looks like the evapotranspiration chart above replicates the quirk noted here whereby a tick value's shown 2x since we're enforcing a set number of tick values, but the range [0.0..0.2] only has 3 to display.

**Comment by rajadain** [comment]
I ended up removing the top label in the charts:

image

Also, Safari issues should now be fixed. Tried to fix IE11 issues, but that was more complicated, so I'm leaving it for #2098.

It should be noted that the hover tips are inaccurate because they are rounded. We should have more accurate values in a future card that should make the tooltips look like the mockup.

**Comment by rajadain** [comment]
@azavea-bot rebuild

**Comment by kellyi** [comment]
Tested this out and generally working well in Chrome, FF, and Safari! I noticed two immediate quirks which I think we should handle elsewhere:

- the stacked bar chart initially shows up as non-stacked on all browsers; subsequently it becomes a stacked chart, and I think this may be worth dealing with when we address the loading state:

screen shot 2017-08-17 at 10 08 29 am

I'm not sure what part of this was obviating this but I think something about the `if (!stacked)... else` check may have guarded against the change.

- in some cases I'm able to hit the back button in the browser and have the compare view show above the Projects list:

screen shot 2017-08-17 at 10 12 33 am

Not sure what causes this but we should make an issue to handle fixing it.

**Comment by rajadain** [comment]
The most recent two commits should take care of the mentioned issues:

**Comment by rajadain** [comment]
Thanks for reviewing! Going to squash and merge shortly.

**Pull request merged**